pkg/uninstaller/deletion.go

```diff
@@ -43,13 +46,68 @@ func DisableBroadcasting(client dynamic.Interface) error {
 	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(t.UnstructuredContent(), &clusterconfigs); err != nil {
 		return err
 	}
-	klog.V(5).Infof("Patching ClusterConfigs")
-	for _, item := range clusterconfigs.Items {
-		patch := []byte(`{"spec": {"advertisementConfig": { "outgoingConfig" : { "enableBroadcaster" : false}}}}`)
-		_, err = r1.Namespace(item.Namespace).Patch(context.TODO(), item.Name, types.MergePatchType, patch, metav1.PatchOptions{})
-		if err != nil {
-			return err
-		}
+	if len(clusterconfigs.Items) != 1 {
+		return fmt.Errorf("ERROR: Wrong number of ClusterConfig: %v", len(clusterconfigs.Items))
+	}
+	err = patchClusterConfig(forgeUninstallClusterConfig, &clusterconfigs.Items[0], client)
+	return err
+}
+
+// DeleteAllForeignClusters deletes all ForeignCluster resources.
+func DeleteAllForeignClusters(client dynamic.Interface) error {
```

Comment:
Could be worth using proper contexts instead of TODO() in order to cleanly react to termination signals